src/parser.rs

```diff
@@ -81,28 +81,24 @@ pub fn parse_command(lexer: &mut Lexer, script: &mut Script) -> Result<bool, (Pa
 			"position" => {
 				let instance = InstanceName(inline(lexer.string())?);
 				let position = position(lexer)?;
-				let animation = animation(lexer, true)?;
+				let animation = animation(lexer)?;
 				script.commands.push(Command::Position(instance, position, animation));
 			}
 			"spawn" => {
 				let character = CharacterName(inline(lexer.string())?);
 				let state = StateName(inline(lexer.string())?);
 				let position = position(lexer)?;
-				let mut check_with = true;
 				let mut is_end = false;
-				let instance_name = match inline(lexer.token())? {
+				let instance_name = match inline(lexer.peek())? {
```

Comment:
Most of this logic can probably be deleted with a check on the peeked token. As an outline:
1. Check if peeked token is a string
2. If it is a string, consume the token and parse it as `Some` instance name
You don't need to check if it's the end anymore as the animation function works regardless of this fact.